### ISSUE TITLE
Usage of Control component with combination of state changes not allowing dropdown to close when clicking outside 

### ISSUE THREAD

**Issue by deflexor** [JedWatson/react-select#5500]
Hi

Here i an example https://codesandbox.io/s/codesandboxer-example-forked-1vmp7q?file=/example.tsx

Here is very simple `Control` component
```
  const Control = ({
    children,
    ...props
  }: ControlProps<ColourOption, false>) => {
    return <components.Control {...props}>{children}</components.Control>;
  };
```

Here is `useState`
```
const [isRowHover, setRowHover] = useState<string | null>(null);
```

And here is Select that we render:

```
      <div
        onMouseEnter={() => setRowHover(true)}
        onMouseLeave={() => setRowHover(false)}
      >
        <Select
          options={colourOptions}
          components={{
            Control
          }}
        />
      </div>
```
If you remove `Control` from `components` or remove any of `onMouse...`, then `Select` works just ok, but in this setup it behaves 
differenlty and you can't close it by clicking outside of it.

**Comment by Rall3n** [comment]
Hello @deflexor,

please read our documentation on ["Defining components"](https://react-select.com/components#defining-components).

**Tl;dr:** Components should not be defined inside the render method of another component, because that can cause a loss of state (e.g. input focus). Always define components outside of the rendering scope:

```jsx
// Bad
const MySelect = () => {
  const CustomControl = () => {};

  return <Select components={{ Control: CustomControl }} />
}

// Good
const CustomControl = () => {};

const MySelect = () => {
  return <Select components={{ Control: CustomControl }} />
}
```

-----

Closing as not a problem with the library, but with React rendering in general.
